api/envoy/extensions/filters/http/adaptive_concurrency/v3/adaptive_concurrency.proto

```diff
@@ -105,4 +106,12 @@ message AdaptiveConcurrency {
   // If set to false, the adaptive concurrency filter will operate as a pass-through filter. If the
   // message is unspecified, the filter will be enabled.
   config.core.v3.RuntimeFeatureFlag enabled = 2;
+
+  // This field allows for a custom HTTP response status code to the downstream client when
+  // the concurrency limit has been exceeded.
+  // Defaults to 503 (Service Unavailable).
+  //
+  // .. note::
+  //   If this is set to < 400, 503 will be used instead.
+  type.v3.HttpStatus concurrency_limit_exceeded_status = 3;
```

Comment:
Sorry, one more question that I missed before, why is this in `AdaptiveConcurrency` and not in `GradientControllerConfig`?

Reply:
I tried to keep adaptive concurrency consistent with `RateLimit`. The 503 was returned by `AdaptiveConcurrency` so I thought it made sense to configure the status code at this level. I got the impression that the `GradientControllerConfig` was related more to calculating the value than handling the response.

Reply:
I'm happy either way, made sense for me to put it here as that's where the existing 503 was returned but a second opinion can't hurt

Reply:
I looked at refactoring, at the moment the `GradientController` doesn't have any HTTP level concepts in it. It seems a bit strange to me to  introduce it for this use case

Reply:
The way I read it, it seems that GradientController is currently a type of HTTP adaptive controller (could be others in the future), and that it imposes the limits.
As it also defines the concurrency limits, it may also define the return val.
That said, we can probably have this field as a filter level setting, like you are suggesting, because it points out to a configuration of the HTTP filter side of things. I guess we can have an override for this in the future, if additional controllers are added.